base/src/org/compiere/model/MTable.java

```diff
@@ -876,22 +878,13 @@ public String getSQLCreate()
 	 *	@return int retValue
 	 */
 	public static int getTable_ID(String tableName) {
-		int retValue = 0;
-		String SQL = "SELECT AD_Table_ID FROM AD_Table WHERE tablename = ?";
-		PreparedStatement pstmt = null;
-		ResultSet rs = null;
-		try {
-			pstmt = DB.prepareStatement(SQL, null);
-			pstmt.setString(1, tableName);
-			rs = pstmt.executeQuery();
-			if (rs.next())
-				retValue = rs.getInt(1);
-		} catch (Exception e) {
-			s_log.log(Level.SEVERE, SQL, e);
-			retValue = -1;
-		} finally {
-			DB.close(rs, pstmt);
+		Integer tableId = s_tableNameCache.get(tableName);
+		if(Optional.ofNullable(tableId).orElse(new Integer(0)) > 0) {

```

Comment:
If using the Optional pattern, you could use .OrElseGet and do it all in one line

    public static int getTable_ID(String tableName) {

        return Optional.ofNullable(s_tableNameCache.get(tableName))
                .orElseGet(() -> 
                    getTable_IDAndAddToCache(tableName)
                );

    }

    private static Integer getTable_IDAndAddToCache(String tableName) {

        int retValue = DB.getSQLValue(null,
                "SELECT AD_Table_ID FROM AD_Table WHERE tablename = ?",
                tableName);
        s_tableNameCache.put(tableName, retValue);
        return retValue;

    }

Would the transaction name ever be required here?

Reply:
@mckayERP is right is not same orElse that  orElseGet to runtime the first is evaluate and second only if get None